src/replication/state.cpp

```diff
@@ -226,8 +229,70 @@ bool ReplicationState::HandleVersionMigration(durability::ReplicationRoleEntry &
       [[fallthrough]];
     }
     case durability::DurabilityVersion::V3: {
-      // Real migration is happening in JSON functions, here just bump version to V4.
+      std::map<std::string, std::string> to_put;
```

Comment:
> Seems like the KVStore first does the Puts and then Deletes, so some keys might get deleted by accident.
I think this is mistake, we should first Delete and then Put because RocksDB applies operations in order

Reply:
I will just use PutMultiple